KaitaiStream.cs

```diff
@@ -7,67 +7,89 @@
 namespace Kaitai
 {
     /// <summary>
-    /// The base Kaitai steam which exposes an API for the Kaitai Struct framework.
+    /// The base Kaitai stream which exposes an API for the Kaitai Struct framework.
     /// It's based off a <code>BinaryReader</code>, which is a little-endian reader.
     /// </summary>
     public partial class KaitaiStream : BinaryReader
     {
         #region Constructors
 
+        /// <summary>
+        /// Create a KaitaiStream backed by abstract stream. It could be in-memory buffer or open file.
+        /// </summary>
         public KaitaiStream(Stream stream) : base(stream)
         {
         }
 
-        ///<summary>
-        /// Creates a KaitaiStream backed by a file (RO)
-        ///</summary>
-        public KaitaiStream(string file) : base(File.Open(file, FileMode.Open, FileAccess.Read, FileShare.Read))
+        /// <summary>
+        /// Create a KaitaiStream backed by a closed file (in read-only binary-mode).
+        /// </summary>
+        public KaitaiStream(string filename) : base(File.Open(filename, FileMode.Open, FileAccess.Read, FileShare.Read))
         {
         }
 
-        ///<summary>
-        ///Creates a KaitaiStream backed by a byte buffer
-        ///</summary>
-        public KaitaiStream(byte[] bytes) : base(new MemoryStream(bytes))
+        /// <summary>
+        /// Create a KaitaiStream backed by in-memory buffer.
+        /// </summary>
+        public KaitaiStream(byte[] data) : base(new MemoryStream(data))
         {
         }
 
+        /// <summary>
+        /// Used internally.
+        /// </summary>
         private ulong Bits = 0;
+        /// <summary>
+        /// Used internally.
+        /// </summary>
         private int BitsLeft = 0;
 
+        /// <summary>
+        /// Used internally.
+        /// Caches the current plaftorm endianness and allows emited bytecode to be optimised. Thanks to @Arlorean.
+        /// </summary>
```

Comment:
Probably warrants a `<remarks>...</remarks>` section with a longer explanation of how this works, or at least a pointer to that discussion.

Reply:
Added link, thanks, fixed.